communications/signals.py

```diff
@@ -0,0 +1,70 @@
+from django.db.models.signals import post_save, post_migrate
+from django.dispatch import receiver
+from django.conf import settings
+from .models import UserNotificationPreference, NotificationType, UserNotificationTypePreference
+
+
+@receiver(post_save, sender=settings.AUTH_USER_MODEL)
+def create_user_notification_preferences(sender, instance, created, **kwargs):
+    """
+    Create default notification preferences and type preferences when a new user is created.
+    """
+    if created:
+        preference = UserNotificationPreference.objects.create(user=instance)
+        
+        notification_types = NotificationType.objects.filter(is_active=True)
+        for notification_type in notification_types:
+            UserNotificationTypePreference.objects.create(
+                user_preference=preference,
+                notification_type=notification_type,
+                frequency=notification_type.default_frequency
+            )
+
+
+@receiver(post_migrate)
+def create_initial_notification_types(sender, **kwargs):
+    """
+    Create initial notification types after migrations.
+    This ensures the types exist even if the data migration wasn't run.
+    """
+    if sender.name == 'communications':
+        from django.db import transaction
+        
+        notification_types = [
+            {
+                'code': 'startup_saved',
+                'name': 'Startup Saved',
+                'description': 'Notification when a user saves a startup to their favorites',
+                'default_frequency': 'immediate',
+            },
+            {
+                'code': 'project_followed',
+                'name': 'Project Followed',
+                'description': 'Notification when a user follows a project',
+                'default_frequency': 'immediate',
+            },
+            {
+                'code': 'message_received',
+                'name': 'Message Received',
+                'description': 'Notification when a user receives a new message',
+                'default_frequency': 'immediate',
+            },
+            {
+                'code': 'project_updated',
+                'name': 'Project Updated',
+                'description': 'Notification when a followed project is updated',
+                'default_frequency': 'daily_digest',
+            },
+        ]
```

Comment:
The notification_types list is hardcoded. If notification types are expected to change or be configurable, consider loading from a fixture or settings.